resolver/parallel_best_resolver.go

```diff
@@ -38,15 +38,19 @@ type requestResponse struct {
 
 // NewParallelBestResolver creates new resolver instance
 func NewParallelBestResolver(upstreamResolvers map[string][]config.Upstream, bootstrap *Bootstrap) (Resolver, error) {
+	logger := logger("parallel resolver")
 	s := make(map[string][]*upstreamResolverStatus, len(upstreamResolvers))
 
 	for name, res := range upstreamResolvers {
 		resolvers := make([]*upstreamResolverStatus, len(res))
 
+		var errResolvers int
+
 		for i, u := range res {
 			r, err := NewUpstreamResolver(u, bootstrap)
 			if err != nil {
-				return nil, err
+				logger.Warnf("upstream group %s: %v", name, err)
```

Comment:
slice in line 41 will be initialized with resolver count. If one resolver throws error, slice will contain a nil resolver. I can reproduce it with following config:
```yaml
upstream:
  default:
  - https://wrong.url1
  - https://wrong.url2
  - https://dns.google/resolve
```

Reply:
Correct, will fix this.